Missing "Manifest" from glossary k8s.io/docs/reference/glossary/ #13546 (#16193)
kubernetes/website: content/en/docs/reference/glossary/manifest.md

---
title: Manifest
id: manifest
date: 2019-06-28
short_description: >

>   A serialized definition of one or more Kubernetes API objects.

Reviewer: Maybe s/definition/specification/. I think the key here is that it's a description of desired state without any status. Definition is conflated with structure due to CRDs.

Author: I have replaced definition with specification.

>   A file containing the serialized definition of one or more Kubernetes API objects.

Reviewer 2: Remember @steveperry-53's feedback that a manifest doesn't have to be a file.
```suggestion
  A serialized definition of one or more Kubernetes API objects.
```

>   Manifest is a Kubernetes "API object description" written as config file (JSON or YAML). 

Reviewer 3: ```suggestion
   Definition of a Kubernetes API object, serialized into a configuration file (eg JSON or YAML)
```

>   Definition of a Kubernetes API object, serialized into a configuration file (eg JSON or YAML)

Reviewer 4: I'd almost simplify this further to just

"A file containing the serialized definition of one or more Kubernetes API objects"

I don't think it's particularly useful to say "configuration file" versus just "file".

aka:
tags:
- fundamental
---

> Manifest is a Kubernetes "API object description" written as config file (JSON or YAML).

Reviewer 3: @steveperry-53 [suggested](https://github.com/kubernetes/website/pull/16193/files#r329233744) “Definition of a Kubernetes API object in JSON or YAML format.” which is also a good one to use here.

<!--more-->

> Manifest defines the desired state of the object that Kubernetes will maintain upon object creation. A single file can contain muntiple manifests. 

Reviewer 3: ```suggestion
A manifest defines the desired state of object that Kubernetes will maintain when you apply the manifest. Each configuration file can contain multiple manifests.
```

> A manifest defines the desired state of object that Kubernetes will maintain when you apply the manifest. Each configuration file can contain multiple manifests.

Reviewer 4: s/configuration file/file/

s/multiple manifests/multiple API objects/

Reviewer 5: I think it's a manifest even if it isn't saved as a file.
Suggestion:
Definition of a Kubernetes API object in JSON or YAML format.

Reviewer 4: Yup, that's better Steve. ++

Reviewer 3: How about:
```suggestion
 You can use a manifest to define an object that Kubernetes will maintain when you apply the manifest. You can define multiple API objects within a single file.
```

I would avoid mentioning _desired state_ here because that term can also refer more narrowly to the `spec` field within an object.

Reviewer 3: You can use a manifest to define an object that doesn't have a desired state, eg Secret. As soon as you define a Secret there's no additional action that needs to happen to make the Secret valid / usable.

> A manifest defines the desired state of object that Kubernetes will maintain when you apply the manifest. Each configuration file can contain multiple manifests.

Reviewer 5: "state of object" -> "state of an object"

---
title: Manifest
id: manifest
date: 2019-06-28

> full_link: /docs/user-journeys/users/application-developer/foundational/#section-2

Reviewer 3: I think I would omit this link rather than hyperlink to that page, to be honest.

It doesn't seem like it's a stable link, in as much as the contents of `#section-2` might be something else in the future.

Author: I have made necessary update in the commit below.

short_description: >
  Manifest is a Kubernetes "API object description" written as config file (JSON or YAML) file. A file as contain one or more manifests. 

aka:
tags:
- fundamental
---
Manifest is a Kubernetes "API object description" written as config file (JSON or YAML).

<!--more-->

> Configuration files - Written in YAML or JSON, these files describe the desired state of your application in terms of Kubernetes API objects. A file can include one or more API object descriptions (manifests).

Reviewer 3: This reads like a (narrow) definition of “configuration file”.

Author: I have made necessary update in the commit below.
